File: tests/fixtures/java_hello/src/test/java/com/example/hello/GreeterTest.java
package com.example.hello;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class GreeterTest {

    @Test
    void greetsCasuallyByDefault() {
        Greeter g = Greeter.make("Ada");
        assertEquals("Hello, Ada!", g.greet());
    }

    @Test
    void supportsFormalGreeting() {
        Greeter g = new Greeter("Grace", true);
        assertEquals("Good day, Grace!", g.greet());
    }

    @ParameterizedTest
    @ValueSource(strings = {"Linus", "Hopper"})
    void saysGoodbye(String name) {
        assertTrue(new Greeter(name).goodbye().endsWith("!"));
    }
}

File: tests/fixtures/typescript_hello/tests/greeter.test.ts
import { describe, it, test, expect } from "vitest";
import { Greeter, makeGreeter } from "../src/greeter";

describe("Greeter", () => {
  it("greets casually by default", () => {
    expect(makeGreeter("Ada").greet()).toBe("Hello, Ada!");
  });

  test("supports formal greeting", () => {
    const g = new Greeter({ name: "Grace", formal: true });
    expect(g.greet()).toBe("Good day, Grace!");
  });

  it("says goodbye", () => {
    expect(makeGreeter("Linus").goodbye()).toBe("Bye, Linus!");
  });
});
